kfs-core/src/main/java/edu/arizona/kfs/fp/document/validation/impl/NonCheckDisbursementRequiredReferenceFieldValidation.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * The Kuali Financial System, a comprehensive financial management system for higher education.
+ * 
+ * Copyright 2005-2014 The Kuali Foundation
+ * 
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package edu.arizona.kfs.fp.document.validation.impl;
+
+import edu.arizona.kfs.gl.service.GlobalTransactionEditService;
+import org.apache.commons.lang.StringUtils;
+import org.kuali.kfs.fp.document.NonCheckDisbursementDocument;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.KFSKeyConstants;
+import org.kuali.kfs.sys.Message;
+import org.kuali.kfs.sys.businessobject.AccountingLineBase;
+import org.kuali.kfs.sys.document.validation.event.AttributedDocumentEvent;
+import org.kuali.rice.krad.datadictionary.BusinessObjectEntry;
+import org.kuali.rice.krad.util.GlobalVariables;
+
+import static org.kuali.kfs.sys.KFSPropertyConstants.REFERENCE_NUMBER;
+
+
+public class NonCheckDisbursementRequiredReferenceFieldValidation extends org.kuali.kfs.fp.document.validation.impl.NonCheckDisbursementRequiredReferenceFieldValidation {
+
+    private GlobalTransactionEditService globalTransactionEditService;
+
+
+    @Override
+    public boolean validate(AttributedDocumentEvent event) {
+
+        NonCheckDisbursementDocument document = (NonCheckDisbursementDocument) event.getDocument();
+
+        boolean valid = true;
+        Class clazz = null;
+        BusinessObjectEntry boe;
+
+        if (getAccountingLineForValidation().isSourceAccountingLine()) {
+            clazz = document.getSourceAccountingLineClass();
+        } else if (getAccountingLineForValidation().isTargetAccountingLine()) {
+            clazz = document.getTargetAccountingLineClass();
+        }
+
+        boe = getDataDictionaryService().getDataDictionary().getBusinessObjectEntry(clazz.getName());
+
+        if (StringUtils.isEmpty(getAccountingLineForValidation().getReferenceNumber())) {
+            putRequiredPropertyError(boe, REFERENCE_NUMBER);
+            valid = false;
+        }
+
```

Comment:
Why not execute validate on super and then add the GTE specific validation? Seems like lines 43-60 are just a duplication of the overridden super method code.